apis/changelogs/proto/v1alpha1/changelog.proto

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2024 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+// buf:lint:ignore PACKAGE_DIRECTORY_MATCH
+package changelogs.proto.v1alpha1;
+
+option go_package = "github.com/crossplane/crossplane-runtime/apis/changelogs/proto/v1alpha1";
+
+// ChangeLogService is a service that provides the ability to send change log
+// entries.
+service ChangeLogService {
+  // SendChangeLog sends a change log entry to the change log service.
+  rpc SendChangeLog (SendChangeLogRequest) returns (SendChangeLogResponse) {}
+}
+
+// SendChangeLogRequest represents a request to send a single change log entry.
+message SendChangeLogRequest {
+  // The change log entry to send as part of this request.
+  ChangeLogEntry entry = 1;
+}
+
+// ChangeLogEntry represents a single change log entry, with detailed information
+// about the resource that was changed.
+message ChangeLogEntry {
```

Comment:
One thing I totally spaced on - I think we should record a timestamp client side and send it in the RPC. I see you're currently recording it in the sidecar. It probably doesn't make a huge difference, but given we'll tolerate a delay of up to 10 seconds when sending the RPC, recording the timestamp in the sidecar could cause changes to appear to happen slightly later than they actually did.

I see there's a well-known type we can use: https://protobuf.dev/reference/protobuf/google.protobuf/#timestamp

Reply:
yep that is a good idea for this scenario. Typically server side timestamps are used, but in this scenario with just one client and a potential sending delay, it make sense to standardize on the client side. 

Done in https://github.com/crossplane/crossplane-runtime/pull/754/commits/b3b7e6da82e2823c45b552e186ef6bfb8277110d